src/test/run-make/issue-14698/Makefile

```diff
@@ -0,0 +1,4 @@
+-include ../tools.mk
+
+all:
+	TMPDIR=fake $(RUSTC) foo.rs 2>&1 | grep "couldn't create a temp dir:"
```

Comment:
One might argue that this is not actually testing the real meat of the PR -- to do that I'd say you would want to grep for `fake` too.

but that is such a minor nit that I am not going to hold up this PR on it